lib/text/simple_text_displayer.js

```diff
@@ -22,66 +26,246 @@ goog.require('shaka.text.Utils');
  */
 shaka.text.SimpleTextDisplayer = class {
   /**
-   * @param {HTMLMediaElement} video
-   * @param {string} label
+   * @param {shaka.Player} player
    */
-  constructor(video, label) {
-    /** @private {HTMLMediaElement} */
-    this.video_ = video;
+  constructor(player) {
+    /** @private {?shaka.Player} */
+    this.player_ = player;
 
-    /** @private {string} */
-    this.textTrackLabel_ = label;
+    /** @private {?HTMLMediaElement} */
+    this.video_ = null;
 
-    /** @private {TextTrack} */
-    this.textTrack_ = null;
+    /** @private {!Map<number, !HTMLTrackElement>} */
+    this.trackNodes_ = new Map();
+
+    /** @private {number} */
+    this.trackId_ = -1;
+
+    /** @private {boolean} */
+    this.visible_ = false;
+
+    /** @private {boolean} */
+    this.hasMicroTask_ = false;
+
+    /** @private */
+    this.onLoaded_ = () => {
+      this.player_.removeEventListener(
+          shaka.util.FakeEvent.EventName.Loaded,
+          this.onLoaded_,
+      );
+      if (
+        this.player_.getLoadMode() === 2 // shaka.Player.LoadMode.MEDIA_SOURCE
+      ) {
```

Comment:
```suggestion
      // shaka.Player.LoadMode.MEDIA_SOURCE
      if (this.player_.getLoadMode() === 2) {
```

We prefer this format in all if-else statements.